src/wrapper/jnienv.rs

```diff
@@ -303,8 +303,17 @@ impl<'a> JNIEnv<'a> {
         Ok(jni_call!(self.internal, NewLocalRef, obj.into_inner()))
     }
 
-    #[allow(dead_code)]
-    fn delete_local_ref(&self, obj: JObject) -> Result<()> {
+
+    /// Deletes the local reference.
+    ///
+    /// Local references are valid for the duration of a native method call. They are
+    /// freed automatically after the native method returns. Each local reference costs
+    /// some amount of Java Virtual Machine resource. Programmers need to make sure that
+    /// native methods do not excessively allocate local references. Although local
+    /// references are automatically freed after the native method returns to Java,
+    /// excessive allocation of local references may cause the VM to run out of memory
+    /// during the execution of a native method.
```

Comment:
Also, when JNI invocation API is used, the runtime will not free local references automatically until [the thread detaches](https://developer.android.com/training/articles/perf-jni.html#local_and_global_references):

> If you attach a native thread with AttachCurrentThread, the code you are running will never automatically free local references until the thread detaches. Any local references you create will have to be deleted manually.